---------------------
PatchSet 11036 
Date: 2006/09/25 19:31:34
Author: serassio
Branch: HEAD
Tag: (none) 
Log:
Bug #1776: squid -2.6.STABLE4-20060925 doesn't compile properly

n_coss_dirs is referenced from both coss and aufs, this errror happens
when building with aufs only. To avoid this, n_coss_dirs must be in globals.c

Members: 
	src/globals.h:1.121->1.122 
	src/fs/aufs/async_io.h:1.2->1.3 
	src/fs/coss/store_dir_coss.c:1.59->1.60 

Index: squid/src/globals.h
===================================================================
RCS file: /cvsroot/squid/squid/src/globals.h,v
retrieving revision 1.121
retrieving revision 1.122
diff -u -r1.121 -r1.122
--- src/globals.h	30 Jul 2006 23:27:03 -0000	1.121
+++ src/globals.h	25 Sep 2006 19:31:34 -0000	1.122
@@ -1,6 +1,6 @@
 
 /*
- * $Id: globals.h,v 1.121 2006/07/30 23:27:03 hno Exp $
+ * $Id: globals.h,v 1.122 2006/09/25 19:31:34 serassio Exp $
  *
  *
  * SQUID Web Proxy Cache          http://www.squid-cache.org/
@@ -177,5 +177,6 @@
 extern int need_linux_tproxy;	/* 0 */
 #endif
 extern int opt_parse_cfg_only;	/* 0 */
+extern int n_coss_dirs;		/* 0 */
 
 #endif /* SQUID_GLOBALS_H */
Index: squid/src/fs/aufs/async_io.h
===================================================================
RCS file: /cvsroot/squid/squid/src/fs/aufs/async_io.h,v
retrieving revision 1.2
retrieving revision 1.3
diff -u -r1.2 -r1.3
--- src/fs/aufs/async_io.h	29 Jul 2006 17:37:18 -0000	1.2
+++ src/fs/aufs/async_io.h	25 Sep 2006 19:31:34 -0000	1.3
@@ -8,7 +8,6 @@
 #define __ASYNC_IO_H__
 
 extern int n_asyncufs_dirs;
-extern int n_coss_dirs;
 extern int squidaio_nthreads;
 extern int squidaio_magic1;
 extern int squidaio_magic2;
Index: squid/src/fs/coss/store_dir_coss.c
===================================================================
RCS file: /cvsroot/squid/squid/src/fs/coss/store_dir_coss.c,v
retrieving revision 1.59
retrieving revision 1.60
diff -u -r1.59 -r1.60
--- src/fs/coss/store_dir_coss.c	23 Sep 2006 10:34:41 -0000	1.59
+++ src/fs/coss/store_dir_coss.c	25 Sep 2006 19:31:35 -0000	1.60
@@ -1,6 +1,6 @@
 
 /*
- * $Id: store_dir_coss.c,v 1.59 2006/09/23 10:34:41 serassio Exp $
+ * $Id: store_dir_coss.c,v 1.60 2006/09/25 19:31:35 serassio Exp $
  *
  * DEBUG: section 47    Store COSS Directory Routines
  * AUTHOR: Eric Stern
@@ -48,7 +48,6 @@
 #define STORE_META_BUFSZ 4096
 #define HITONLY_BUFS 2
 
-int n_coss_dirs = 0;
 int max_coss_dir_size = 0;
 /* static int last_coss_pick_index = -1; */
 int coss_initialised = 0;
